src/init.c

```diff
@@ -378,6 +383,51 @@ shmem_internal_heap_postinit(void)
               shmem_internal_data_base, shmem_internal_data_length);
 
 #ifdef HAVE_SCHED_GETAFFINITY
+#ifdef USE_HWLOC
+    ret = hwloc_topology_init(&shmem_internal_topology);
+    if (ret < 0) {
+        RETURN_ERROR_MSG("hwloc_topology_init failed (%s)\n", strerror(errno));
+    }
+
+    ret = hwloc_topology_set_io_types_filter(shmem_internal_topology, HWLOC_TYPE_FILTER_KEEP_ALL);
+	if (ret < 0) {
+        RETURN_ERROR_MSG("hwloc_topology_set_io_types_filter failed (%s)\n", strerror(errno));
+    }
+
+    ret = hwloc_topology_load(shmem_internal_topology);
+    if (ret < 0) {
+        RETURN_ERROR_MSG("hwloc_topology_load failed (%s)\n", strerror(errno));
+    }
+#if defined HWLOC_ENFORCE_SINGLE_SOCKET || defined HWLOC_ENFORCE_SINGLE_NUMA_NODE
```

Comment:
Style thing: SOS always puts parentheses around args to `defined`. (But I don't think it matters in this case...?)
```suggestion
#if defined(HWLOC_ENFORCE_SINGLE_SOCKET) || defined(HWLOC_ENFORCE_SINGLE_NUMA_NODE)
```